packages/apollo-server-codegen/ARCHITECTURE.md

```diff
@@ -0,0 +1,67 @@
+# Architecture
+
+## Overview
+
+The code generator is a set of modules designed to accept a GraphQL schema, and from it generate type definitions in TypeScript format (other languages may be supported in the future).
+
+Similar to projects like [graphql code generator](https://graphql-code-generator.com), the generated type definitions describe both the type of the GraphQL objects represented in the schema, and the type of the `resolvers` function that an Apollo Server serving would require to serve the schema.
+
+However, this project expands upon existing efforts by providing accurate type definitions for federated schemas, built using the recently announced Apollo Federation platform.
+
+This package serves only as an API, to interface with it we suggest either the `Apollo CLI`, where it is available as `apollo server:codegen`, or the `[.....]` plugin we have built for `graphql-code-generator`.
+
+## Design
+
+The generator is built using a two-layered approach, where the frontend is able to convert an SDL-formatted input string to an internal representation (`IR`), and one or backends are able to convert that `IR` to files of type declaration source code in a particular programming language. Currently only a backend for TypeScript is provided.
```

Comment:
I'd like to suggest a different nomenclature here, since frontend/backend have pretty well-defined meanings already. I'd remap backend -> consumer, but not sure what to remap frontend to. Something related to normalization, translation, or parsing?

Reply:
This is consistent with the nomenclature compilers use, so I think its reasonable to use it here in ARCHITECTURE.md, but maybe not in something like README.md.